### DISCUSSION TITLE
大佬为何这个网站没法添加元素进去，比如BUTTON

### DISCUSSION THREAD

**Discussion by hackhase** [lisonge/vite-plugin-monkey#121] [General]
https://www.doc88.com/p-3117685065644.html，这个里面没法添加，插件没法运行。
https://www.doc88.com而这个上面却可以，大佬帮忙看看啦！

**Comment by lisonge** [comment]
此域名下的全局对象 Map 被修改为非规范类型导致 vite 内部 Map 方法报错

在 devtools 网络面板阻止此 URL <https://static.doc88.com/assets/js/commonjs.js> 即可

**Reply by hackhase** [reply]
那能不能在app.vue文件里面，用代码来阻止这个URL呢？如果可以的话，代码是怎么样的，谢谢！

**Comment by hackhase** [comment]
那能不能在app.vue文件里面，用代码来阻止这个URL呢？如果可以的话，代码是怎么样的，谢谢！

**Reply by lisonge** [reply]
在js/ vue 里面不能，它是直接写在 html 通过 script 加载的

不过你可以使用 <https://www.tampermonkey.net/documentation.php?locale=zh#meta:webRequest> 将这个链接阻止或者直接重定向到你二次修改过的脚本，在这个脚本里修改源代码将 Map 改成其他名称就行

**Reply by hackhase** [reply]
大佬，我用网上的代码
`chrome.webRequest.onBeforeRequest.addListener((data)=>{
console.log(data);
});`
为何老是报错：Uncaught TypeError: Cannot read properties of undefined (reading 'onBeforeSendHeaders')，帮忙看下，万分感谢！

**Reply by hackhase** [reply]
我按文档写了下
`GM_webRequest([
    { selector: 'res.doc88.com/assets/js/commonjs.js?v=3.76', action: 'cancel' },
], function(info, message, details) {
    console.log(info, message, details);
});`，没法打印出结果
